js/main.js

```diff
@@ -10,49 +15,114 @@ $xhrRandom.addEventListener('load', loading);
+  $imgDiv.className = 'col-full col-50 shadow';
+  var $img = document.createElement('img');
+  $img.setAttribute('src', obj.imgUrl);
+  var $infoColDiv = document.createElement('div');
+  $infoColDiv.className = 'col-full col-50 info-col-padding align-self-start';
+  var $rowDiv = document.createElement('div');
+  $rowDiv.className = 'row';
+  var $titleHeader = document.createElement('h4');
+  $titleHeader.className = 'title-font';
+  $titleHeader.textContent = obj.title;
+
+  $rowDiv.appendChild($titleHeader);
+  $infoColDiv.appendChild($rowDiv);
+  $imgDiv.appendChild($img);
+  $wrapperDiv.appendChild($imgDiv);
+  $wrapperDiv.appendChild($infoColDiv);
+  $mainDiv.appendChild($wrapperDiv);
+
+  return $mainDiv;
+
+}
 $xhrRandom.send();
 // end of feature 1
+
+// feature 2
+var $viewRandom = document.querySelector('.random-view');
+var $viewRecommend = document.querySelector('.genre-view');
+window.addEventListener('DOMContentLoaded', loadview);
+
+function loadview(event) {
+  if (data.view === 'random-view') {
+    $viewRandom.className = 'random-view';
+    $viewRecommend.className = 'genre-view hidden';
+  } else if (data.view === 'genre-view') {
+    $viewRandom.className = 'random-view hidden';
+    $viewRecommend.className = 'genre-view';
+  }
+}
+
+// feature 2 to render genre list
+var $selectGenre = document.getElementById('genre');
+$selectGenre.addEventListener('change', selectGenre);
+
+function selectGenre(event) {
+  $genreDivEl.replaceChildren();
+  var $genreID;
+  var $genreHeader = document.querySelector('.genre-text');
+  if (event.target.value === 'sliceOfLife') {
+    $genreHeader.textContent = 'Results for Genre: Slice of Life';
+  } else if (event.target.value === 'sciFi') {
+    $genreHeader.textContent = 'Results for Genre: Sci-Fi';
+  } else {
+    $genreHeader.textContent = 'Results for Genre: ' + event.target.value[0].toUpperCase() + event.target.value.substring(1);
+  }
+
+  for (var prop in data.genres[0]) {
+    if (prop === event.target.value) {
+      $genreID = data.genres[0][prop];
+      data.view = 'genre-view';
+      loadview(data.view);
+      break;
+    }
+  }
+
+  var $xhrGenre = new XMLHttpRequest();
+  $xhrGenre.open('GET', 'https://api.jikan.moe/v4/anime?genres=' + $genreID + '&page=1&sfw=true');
+  $xhrGenre.responseType = 'json';
+  $xhrGenre.addEventListener('load', getGenre);
+  var genreObj = {};
+  function getGenre() {
+    for (var i = 0; i < 12; i++) {
+      if ($xhrGenre.response.data[i].images.jpg.large_image_url) {
+        genreObj.imgUrl = $xhrGenre.response.data[i].images.jpg.large_image_url;
+        if ($xhrGenre.response.data[i].title_english) {
+          genreObj.title = $xhrGenre.response.data[i].title_english;
+        } else {
+          genreObj.title = $xhrGenre.response.data[i].title;
+        }
+      } else {
+        genreObj.imgUrl = 'images/placeholder-image-square.jpg';
+      }
+      $genreDivEl.appendChild(renderLists(genreObj));
+    }
+  }
```

Comment:
You shouldn't generally have function definitions inside of other function definitions. If you are doing this to have access to a value from the parent function, you should instead just give this function a parameter and pass that value you need as an argument.